peepingtom/core/datablocks/meshblock.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+
+from ..base import DataBlock
+
+
+class MeshBlock(DataBlock):
+    """
+    Data structure for holding n-dimensional mesh data
+    """
+    def __init__(self, vertices: np.ndarray, faces: np.ndarray, **kwargs):
+        """
+
+        Parameters
+        ----------
+        vertices : array of shape (v, m)
+                   v vertices in m dimensions
+        faces : array of shape (f, 3)
+                array of indices into the vertices defining connectivity
+        kwargs : keyword arguments passed to DataBlock
+        """
+        super().__init__(**kwargs)
+        self.vertices = vertices
+        self.faces = faces
+        self.data = self.vertices
+
+    def _data_setter(self, vertices):
+        self._data = vertices
```

Comment:
I kee thinking that the we got ourselvess stuck with this `data` attribute and it's not necessary nor useful, especially in cases like this or `ParticleBlock` where it's not actually representing the whole data... We need to discuss about it!

Reply:
we discussed this and agreed to drop it, not useful